API.md

```diff
@@ -0,0 +1,518 @@
+  - [POST [base]/Patient](#create-post-pat)
+- [Updating](#update)
+  - [PUT [base]/Patient/[:id]](#update-put-pat)
+- [Searching](#search)
+  - [GET /fhir/r4/Patient?parameter(s)](#search-get)
+
+<a name="cap"/>
+
+## CapabilityStatement
+
+A capability statement is available at `[base]/fhir/r4/CapabilityStatement`:
+
+<a name="cap-get"/>
+
+### GET `[base]/fhir/r4/CapabilityStatement`
+
+```json
+{
+  "status": "active",
+  "kind": "instance",
+  "software": {
+    "name": "Sara Alert",
+    "version": "v1.4.1"
+  },
+  "implementation": {
+    "description": "Sara Alert API"
+  },
+  "fhirVersion": "4.0.1",
+  "format": [
+    "xml",
+    "json"
+  ],
+  "rest": [
+    {
+      "mode": "server",
+      "security": {
+        "cors": true,
+        "service": [
+          {
+            "coding": [
+              {
+                "system": "http://terminology.hl7.org/CodeSystem/restful-security-service",
+                "code": "OAuth"
+              }
+            ]
+          }
+        ]
+      },
+      "resource": [
+        {
+          "type": "Patient",
+          "interaction": [
+            {
+              "code": "read"
+            },
+            {
+              "code": "update"
+            },
+            {
+              "code": "create"
+            },
+            {
+              "code": "search-type"
+            }
+          ],
+          "searchParam": [
+            {
+              "name": "family",
+              "type": "string"
+            },
+            {
+              "name": "given",
+              "type": "string"
+            },
+            {
+              "name": "telecom",
+              "type": "string"
+            },
+            {
+              "name": "email",
+              "type": "string"
+            }
+          ]
+        },
+        {
+          "type": "Observation",
+          "interaction": [
+            {
+              "code": "read"
+            }
+          ]
+        },
+        {
+          "type": "QuestionnaireResponse",
+          "interaction": [
+            {
+              "code": "read"
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  "resourceType": "CapabilityStatement"
+}
+```
+
+<a name="auth"/>
+
+## Authenticating
+
+Authenticate using your Sara Alert account credentials via `/oauth/token`:
+
+<a name="auth-post"/>
+
+### POST `[base]/oauth/token`
+
+#### Request Body
+
+```json
+{
+  "grant_type" : "password",
+  "username" : "state1_epi@example.com",
+  "password" : "1234567ab!"
+}
+```
+
+#### Response
+
+```json
+{
+  "access_token": ">>>> token <<<<",
+  "token_type": "Bearer",
+  "expires_in": 7199,
+  "created_at": 1589332127
+}
+```
+
+Use this token for all subsequent requests, via Authorization header, i.e. `'Authorization': "Bearer >>>> token <<<<"`.
+
+<a name="read"/>
+
+## Reading
+
+The API supports reading monitorees, monitoree lab results, and monitoree daily reports.
+
+<a name="read-get-pat"/>
+
+### GET `[base]/Patient/[:id]`
```

Comment:
Performing a search in a specific patient that does not exist, e.g. `[base]/fhir/r4/Patient/foo` returns a 403 forbidden, and the following JSON:

```json
{
    "error": "Forbidden"
}
```

The response object should be an [OperationOutcome](http://hl7.org/fhir/operationoutcome.html) on HTTP status codes `4XX` or `5XX` or no body. See http://hl7.org/fhir/http.html#Status-Codes

Reply:
Fixed